### PR TITLE
Better: remove Time#succ (obsolete since 1.9.2). Feature #17039

### PR DESCRIPTION
**Pull request by ylecuyer**

### CODE REVIEW

File: spec/ruby/core/time/succ_spec.rb [hunk 1]
```diff
@@ -2,31 +2,33 @@
 require_relative 'fixtures/classes'
```
Comment (ylecuyer) [inline spec/ruby/core/time/succ_spec.rb:2]: See https://github.com/ruby/ruby/pull/3525

File: spec/ruby/core/time/succ_spec.rb [hunk 2]
```diff
@@ -2,31 +2,33 @@
...
 
 describe "Time#succ" do
-  it "returns a new time one second later than time" do
-    suppress_warning {
-      @result = Time.at(100).succ
-    }
+  ruby_version_is ""..."2.6" do
```
Comment (shyouhei) [inline spec/ruby/core/time/succ_spec.rb:5]: What happened to ruby 2.7?

**Review by shyouhei — COMMENTED** [review]
CI failure seems unrelated.

File: spec/ruby/core/time/succ_spec.rb (hunk 2 above)
Comment (ylecuyer) [reply spec/ruby/core/time/succ_spec.rb:5]: Indeed 2.7 should be included, I'll fix this

**Review by phykos — APPROVED** [review]

**Comment by hsbt** [comment]
@phykos Do not submit the approvement each pull-requests. It's noisy for our team.

**Comment by phykos** [comment]
@hsbt ok

**Review by shyouhei — APPROVED** [review]

File: spec/ruby/core/time/succ_spec.rb [hunk 3]
```diff
@@ -2,31 +2,33 @@
 require_relative 'fixtures/classes'
 
 describe "Time#succ" do
-  it "returns a new time one second later than time" do
-    suppress_warning {
-      @result = Time.at(100).succ
-    }
+  ruby_version_is ""..."3.0" do
```
Comment (eregon) [inline spec/ruby/core/time/succ_spec.rb:5]: I would move this outside just above `describe "Time#succ" do`, so it's clear there is no Time#succ at all in Ruby 3.

**Comment by znz** [comment]
Thank you for your contribution.
https://github.com/ruby/ruby/pull/3841 is merged.

**Pull request closed without merging**